src_c/base.c

```diff
@@ -533,6 +533,62 @@ pg_TwoIntsFromObj(PyObject *obj, int *val1, int *val2)
     return 1;
 }
 
+static int
+pg_IntFromObjEx(PyObject *obj, int *val, const char *msg)
+{
+    if (PyFloat_Check(obj)) {
+        *val = (int)PyFloat_AS_DOUBLE(obj);
+        return 1;
+    }
+
+    *val = PyLong_AsLong(obj);
+    if (PyErr_Occurred()) {
+        if (msg) {
```

Comment:
```suggestion
        if (msg && PyErr_ExceptionMatches(PyExc_TypeError)) {
```
We should not overwrite other errors with `TypeError`. We should only reword the message in `TypeError`s if the caller of this function needs to (that is when msg is non-`NULL`